### PR TITLE
Plugin docs.

### LINKED ISSUE #1536: Create plugin development documentation

### PR DESCRIPTION
**Pull request by wwajerowicz**
Fixes #1536 
screencapture-file-Users-wajerowi-code-nexus-docs-target-paradox-site-main-docs-delta-plugins-index-html-2021-05-07-15_17_07

### CODE REVIEW

File: docs/src/main/paradox/docs/delta/plugins/index.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Plugins
+
+Starting from version 1.5, Delta introduces the ability to extend its functionality using plugins. Plugins enable
+developers to add new functionality to Nexus Delta without the need to modify Delta itself. Plugin can introduce various
+new functionalities, including new resource types and indexing capabilities. 
+
+## Plugin development
+
+Plugins used by Delta need to be packaged as a `.jar` file containing the plugin code with all its dependencies.
+Plugins must define exactly one class which extends [PluginDef](https://github.com/BlueBrain/nexus/blob/master/delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/plugin/PluginDef.scala) trait.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/plugins/index.md:10]: All links should be formatted follows:
```scala
@link:[text](link){ open=new }
```

**Review by umbreak — CHANGES_REQUESTED** [review]

**Comment by imsdu** [comment]
It looks good, I think we should emphasize on the fact that it is still experimental and subject to changes in the future as it is still fresh and we may need to change some stuff in the next releases.

**Review by bogdanromanx — COMMENTED** [review]
It's a very good starting point, sufficient for the release; we should improve it with a simple tutorial post release.

File: docs/src/main/paradox/docs/delta/plugins/index.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  - [ReferenceExchange](https://github.com/BlueBrain/nexus/blob/master/delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/ReferenceExchange.scala) - enables Delta to exchange a resource reference for a JSON-LD value allowing Delta to handle multiple resources in a uniform way 
+  - [MetadataContextValue](https://github.com/BlueBrain/nexus/blob/master/delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/MetadataContextValue.scala) - registers metadata context of this plugin into global metadata context 
+  - [RemoteContextResolution](https://github.com/BlueBrain/nexus/blob/master/delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/context/RemoteContextResolution.scala) - enables Delta to resolve static contexts defined by the plugin
+
+### Class loading
+
+In order to avoid clashes between different plugins, Delta uses custom classloader to load plugin classes,
```
[the remark this answers is not in the document]
Comment (bogdanromanx) [reply docs/src/main/paradox/docs/delta/plugins/index.md:70]: I would also recommend adding somewhere that due to the "experimental" status, there are no source/binary compatibility guarantees for the sdks and dependent modules (rdf/sourcing).

File: docs/src/main/paradox/docs/delta/plugins/index.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+which will load classes from the plugin first, then using application classloader and other plugins after that.
+It is therefore recommended to not include in the plugin jar any dependencies which are also provided by SDK.
+
+## Existing plugins
+
+Currently, following Delta functionality is provided by plugins:
+
+- archives @ref:[API Reference](../api/current/kg-archives-api.md) | [code](https://github.com/BlueBrain/nexus/tree/master/delta/plugins/archive/src)
+- SPARQL views @ref:[API Reference](../api/current/views/sparql-view-api.md) | [code](https://github.com/BlueBrain/nexus/tree/master/delta/plugins/blazegraph/src)
+- ElasticSearch views @ref:[API Reference](../api/current/views/elasticsearch-view-api.md) | [code](https://github.com/BlueBrain/nexus/tree/master/delta/plugins/elasticsearch/src)
```
[the remark this answers is not in the document]
Comment (wwajerowicz) [reply docs/src/main/paradox/docs/delta/plugins/index.md:80]: It seems that first letter is capitalised by Elasticsearch https://www.elastic.co/elasticsearch/

**Review by umbreak — APPROVED** [review]

**Review by imsdu — APPROVED** [review]

**Pull request merged**